### DISCUSSION TITLE
Call to undefined function

### DISCUSSION THREAD

**Discussion by Justman100** [InvoicePlane/InvoicePlane#1095] [General]
> Message: Call to undefined function get_setting()

> Filename: application/modules/dashboard/controllers/Dashboard.php

>Line Number: 27

**Comment by Justman100** [comment]
@naui95

**Comment by naui95** [comment]
Hi @Justman100 thank you for the bug report. Could you kindly tell us what were you doing when you encountered the bug?

**Comment by Justman100** [comment]
> Hi @Justman100 thank you for the bug report. Could you kindly tell us what were you doing when you encountered the bug?

- Installed via the Archiv under *[Releases](../releases)*
- First tried with PHP 8.2, then with PHP 8.1

**Comment by naui95** [comment]
> > Hi @Justman100 thank you for the bug report. Could you kindly tell us what were you doing when you encountered the bug?
> 
> - Installed via the Archiv under *[Releases](../releases)*
> - First tried with PHP 8.2, then with PHP 8.1

Are you getting the error in the installation process? Or at an other moment?

**Reply by Justman100** [reply]
Yes, if I want to install it

**Reply by naui95** [reply]
At what point of the installation are you encountering the issue? Do you have a longer stack of the error? What are your logs telling you?

**Comment by nielsdrost7** [comment]
@Justman100 I've converted your bug report to a "discussion" so you can figure out what's going on. The Archiv on the website was made specifically for PHP 8.1
The `get_setting()` function is a very important function within InvoicePlane

It is found in `/application/helpers/settings_helper.php`
If that function cannot be found, that's really strange.

Normally, if I can access the server, I would run `composer dump-autoload -o`
Or I would check that "helpers" directory if that file is there

**Reply by Justman100** [reply]
Hmm, the file exists in said directory

**Reply by nielsdrost7** [reply]
Is this on a server or on your local computer?
How did you change from PHP version 8.2 back to PHP 8.1?

**Reply by Justman100** [reply]
vSever! And yes, it runs PHP 8.1

**Reply by Justman100** [reply]
@nielsdrost7

**Reply by nielsdrost7** [reply]
Go to file `application/modules/dashboard/controllers/Dashboard.php` from your error message.
- See if it is there or maybe badly uploaded.
- scroll to line 27 and look for that function.

"Call to undefined function" should give you a longer error message.
- Look for the file that called this function.
- Set a breakpoint.
A `die("here we are"); is enough for now.
See if you'll get there

Prepare to run `composer install`
